src/backups.py

```diff
@@ -475,11 +498,24 @@ def _pre_restore_checks(self, event: ActionEvent) -> bool:
             event.fail(error_message)
             return False
 
+        logger.info("Checking that this unit was already elected the leader unit")
+        if not self.charm.unit.is_leader():
+            error_message = "Unit cannot restore backup as it was not elected the leader unit yet"
+            logger.warning(error_message)
+            event.fail(error_message)
+            return False
+
```

Comment:
Check needed when scaling down (and the leader is the unit that was removed). It'll be more useful later in the backup/restore tests PR.